DYDX-Bots/Insolvent_Contract/src/agent.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { FindingType, FindingSeverity, Finding, HandleBlock, BlockEvent } from "forta-agent";
+import { createAddress, TestBlockEvent, MockEthersProvider } from "forta-agent-tools/lib/tests";
+import { BigNumber } from "ethers";
+import { provideHandleBlock } from "./agent";
+import BalanceFetcher from "./balance.fetcher";
+import { IMPLEMENTATION_IFACE } from "./utils";
+
+const testProxyAddr: string = createAddress("0xab");
+const testBlockNumbers: number[] = [2, 42, 92, 360];
+const testThreshold: BigNumber = BigNumber.from("3000000000000000000"); // 3
+const testBorrowerDebtBalance: BigNumber[] = [
+  BigNumber.from("5000000000000000000"), // 5
+  BigNumber.from("10000000000000000000"), // 10
+  BigNumber.from("15000000000000000000"), // 15
+  BigNumber.from("20000000000000000000"), // 20
+];
+const testActiveBalanceCurrentEpoch: BigNumber[] = [
+  BigNumber.from("1000000000000000000"), // 1
+  BigNumber.from("9000000000000000000"), // 9
+  BigNumber.from("15000000000000000000"), // 15
+  BigNumber.from("22000000000000000000"), // 22
+];
+
+describe("high tether transfer agent", () => {
+  const mockProvider: MockEthersProvider = new MockEthersProvider();
+  const balanceFetcher: BalanceFetcher = new BalanceFetcher(mockProvider as any, testProxyAddr);
+
+  const createGetTotalBorrowerDebtBalance = (totalBorrowerDebtBalance: BigNumber, blockNumber: number) => {
+    mockProvider.addCallTo(testProxyAddr, blockNumber, IMPLEMENTATION_IFACE, "getTotalBorrowerDebtBalance", {
+      inputs: [],
+      outputs: [totalBorrowerDebtBalance],
+    });
+  };
+
+  const createGetTotalActiveBalanceCurrentEpoch = (totalActiveBalanceCurrentEpoch: BigNumber, blockNumber: number) => {
+    mockProvider.addCallTo(testProxyAddr, blockNumber, IMPLEMENTATION_IFACE, "getTotalActiveBalanceCurrentEpoch", {
+      inputs: [],
+      outputs: [totalActiveBalanceCurrentEpoch],
+    });
+  };
+
+  const handleBlock: HandleBlock = provideHandleBlock(balanceFetcher, testThreshold);
+
+  beforeEach(() => mockProvider.clear());
+
+  it("should detect a contract in an insolvent state", async () => {
```

Comment:
I suggest adding a test case where the bot ignores an insolvency case in a different contract

Reply:
Thinking about, I'm not sure how helpful this will be. Since this isn't `handleTransaction` that detects where an event is emitted, but instead proactively calls a specific contract's function at a specific address that was passed as an argument when the `balanceFetcher` was created. This bot is more proactive, than reactive.

Then because the contract's address was passed as an argument, it wouldn't call the wrong contract.